packages/integration-sdk-cli/src/commands/visualize-types.ts

```diff
@@ -0,0 +1,145 @@
+const collector = (value: string, arr: string[]) => {
+  arr.push(...value.split(','));
+  return arr;
+};
+
+export function visualizeTypes() {
+  return createCommand('visualize-types')
+    .description('Generates a graph of types metadata for all steps')
+    .option(
+      '-p, --project-path <directory>',
+      'Absolute file path to the integration project directory. Defaults to the current working directory.',
+      process.cwd(),
+    )
+    .option(
+      '-g, --graph-file-path <path>',
+      'Absolute file path to the HTML file that should be created/overwritten. Defaults to {CWD}/.j1-integration/types-graph/index.html.',
+    )
+    .option(
+      '-t, --types-file-path <path>',
+      'Absolute file path to the JSON file that should be created/overwritten. Defaults to {CWD}/docs/types.json.',    
+    )
+    .option(
+      '-y, --type <string>',
+      'J1 type(s) to visualize, comma separated if multiple',
+      collector,
+      [],
+    )
+    .action(executeVisualizeTypesAction);
+}
+
+async function executeVisualizeTypesAction(
+  options: VisualizeTypesCommandArgs,
+): Promise<void> {
+  const { projectPath } = options;
+  const types = options.type.length ? undefined : options.type;
+  const graphFilePath =
+    options.graphFilePath ||
+    getDefaultTypesGraphFilePath(projectPath);
+  const typesFilePath = 
+    options.typesFilePath || 
+    getDefaultTypesFilePath(projectPath);
+
+  await executeTypesAction({
+    projectPath,
+    typesFilePath,
+  });
+  
+  log.info('\nLoading types metadata file...\n');
+  const metadataString = await fs.readFile(typesFilePath, {
+    encoding: 'utf-8',
+  });
+
+  const metadata: StepGraphObjectMetadataProperties = JSON.parse(metadataString);
+
+  if (!metadata.entities.length && !metadata.relationships.length) {
+    log.info(
+      'No entities or relationships found to generate types graph for. Exiting.',
+    );
+    return;
+  }
+
+  log.info('\nGenerating local graph from types metadata...\n');
+
+  const edges = getEdgesFromStepRelationshipMetadata(metadata.relationships, { types });
+  const nodes = getNodesFromStepEntityMetadata(metadata.entities, { types, edges });
+  
+
+  const visHtml = generateVisHTML(
+    nodes,
+    edges,
+  )
+
+  await fs.mkdir(path.dirname(graphFilePath), { recursive: true });
+  await fs.writeFile(graphFilePath, visHtml, 'utf-8')
+
+  log.info(`Visualize metadata graph here: ${graphFilePath}`);
+}
+
+function getDefaultTypesGraphFilePath(
+  projectSourceDirectory: string,
+): string {
+  return path.join(projectSourceDirectory, '.j1-integration/types-graph/index.html');
+}
+
+function getNodesFromStepEntityMetadata(
+  entities: StepEntityMetadata[],
+  options?: {
+    types?: string[],
+    edges?: Edge[],
+  }
+): Node[] {
+  let targetTypes: Set<string> | undefined = undefined;
+  if (options?.types !== undefined && options.edges !== undefined) {
+    targetTypes = new Set();
```

Comment:
Minor, but imo it would be slightly cleaner to just move the variable declaration of `targetTypes` into this `if`. This would allow you to avoid the optional chaining in the filter function.